evennia/server/inputfuncs.py

```diff
@@ -654,6 +662,119 @@ def msdp_send(session, *args, **kwargs):
     session.msg(send=((), out))
 
 
+def is_typing_send_update():
+    """
+    Send relevant updates to participants
+
+    """
+    participants = list(_IS_TYPING_PARTICIPANTS.keys())
+
+    for participant in participants:
```

Comment:
The _IS_TYPING_PARTICIPANT never needs to be sent across the wire as far as I can tell, so why do all this conversion back and forth to strings/keys? Just store a dict of sessions  for quick lookup and adding/removal (maybe a `{session: puppet }` map?)